File: topic_5/Python/2-frequencia.py
# Recebe uma lista e um valor n e retorna o total de ocorrências de x

def frequencia(n, xs):
    cont = 0
    for x in xs:
        if x == n:
            cont += 1
    return cont

print(frequencia(1, []) == 0) #True
print(frequencia(4, [4]) == 1) #True
print(frequencia(4, [5]) == 0) #True
print(frequencia(4, [4,4]) == 2) #True
print(frequencia(2, [4,4]) == 0) #True
print(frequencia(5, [4,5,2,1,5,5,9]) == 3) #True

File: topic_5/Haskell/2-frequencia.hs
-- Recebe uma lista e um valor n e retorna o total de ocorrências de x

frequencia n [] = 0
frequencia n xs = sum (map (\x -> 1) (filter (\x -> n == x) xs))

main = do
    a <- readLn :: IO Int
    b <- readLn :: IO [Int]
    print $ frequencia a b